edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/brokerConnection/MqttBrokerConnector.cs

```diff
@@ -166,7 +168,7 @@ public async Task DisconnectAsync()
 
         public async Task<bool> SendAsync(string topic, byte[] payload, bool retain = false)
         {
-            var client = this.mqttClient.Expect(() => new Exception("No mqtt-bridge connector instance found to send messages."));
+            var client = this.mqttClient.Expect(() => new IOException("No mqtt-bridge connector instance found to send messages."));
```

Comment:
IOException is typically used for file I/O.. so it seems a little odd here. Can we use a custom exception here?